internal/encryption/config/config.go

```diff
@@ -73,9 +73,12 @@ type TargetConfig struct {
 //
 // Note: This struct is copied because gohcl does not support embedding.
 type EnforcableTargetConfig struct {
-	Enforced bool           `hcl:"enforced,optional"`
-	Method   hcl.Expression `hcl:"method,optional"`
-	Fallback *TargetConfig  `hcl:"fallback,block"`
+	//TODO if a state{} or plan{} block is defined it implies enforced
```

Comment:
No, enforced has a different purpose. Enforced needs to be there to make sure the state is encrypted even if there is no available method. (e.g. in an environment variable scenario.) This should be explicit to the user, empty blocks can be confusing. An empty state block should not be allowed with this change.

Reply:
We chatted a bit and I'm convinced.  Going to re-work the PR to revert this.